sql/src/main/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitor.java

```diff
@@ -233,8 +239,19 @@ public Projector visitOrderedTopN(OrderedTopNProjection projection, Context cont
         for (int i = numOutputs; i < inputs.size(); i++) {
             orderByIndices[idx++] = i;
         }
+
+        List<DataType> dataTypes = Symbols.typeView(
+            projection.outputs().stream()
+                .filter(s -> s.symbolType() == SymbolType.INPUT_COLUMN ||
```

Comment:
Why do we need to filter for INPUT_COLUMN and LITERAL here?

Reply:
the row columns will be ICs and literals at collection time

Reply:
yes - but why not use simply all? (There shouldn't be any others, or?)

Reply:
at this stage we also have the functions (of ICs)